mhs-reference-implementation/mhs/SDSLookup/tests/test_sdsclient.py

```diff
@@ -0,0 +1,68 @@
+from unittest import TestCase
+import mhs.SDSLookup.sds as sds
+import mhs.SDSLookup.tests.ldap_mocks as mocks
+from utilities.test_utilities import async_test
+
+NHS_SERVICES_BASE = "ou=services, o=nhs"
+
+MHS_OBJECT_CLASS = "nhsMhs"
+
+PARTY_KEY = "A91461-9199084"
+SERVICE_ID = "urn:nhs:names:services:psis:MCCI_IN010000UK13"
+ODS_CODE = "A91461"
+
+
+class TestSDSClient(TestCase):
+
+    @async_test
+    async def test_mhs_details_lookup(self):
+        server, connection = mocks.fake_ldap_server()
+
+        client = sds.SDSClient()
+        client.server = server
+        client.connection = connection
+
+        result = await client._mhs_details_lookup(PARTY_KEY, SERVICE_ID)
+
+        self.assertIsNotNone(result)
+        self.assertEqual(result[0]['attributes']['nhsMHSEndPoint'], ['https://vpn-client-1411.opentest.hscic.gov.uk/'])
```

Comment:
I think this should probably test that the full list of attributes is returned, since it's one of the acceptance criteria on the story. Also, we need to verify that attributes not on the list are filtered.

Reply:
Updated the test to check for all attributes, however because of the mhs attribute call returning 2 sets of attributes if I check the values for an exact match it will fail half the time (due to responses including a unique id). I've included all the attributes bar the two unique ones and still checked that they are present, just not their values. Does this seem reasonable?

Also added length checking to the attribute list to ensure it contains exactly the number we need

Reply:
I think this does the job, but isn't the MHS attribute call returning 2 sets of attributes because that's what we have in the dummy LDAP server data? Couldn't we edit it so that it doesn't return two sets for this query?

Reply:
This is a good point, I've removed the duplicate data and added all attribute checks in so the test should be complete now.